lib/Doctrine/DBAL/DBALException.php

```diff
@@ -282,4 +282,14 @@ public static function typeNotFound($name)
     {
         return new self('Type to be overwritten ' . $name . ' does not exist.');
     }
+
+    public static function invalidColumnIndex(int $index, int $count) : self
```

Comment:
Would it make sense to provide the `$row`, and add the array keys to the message?

Reply:
`$rows` is always a zero-based array since we're fetching a column by index, so the count is enough.

Reply:
:+1: